src/util/audit.c

```diff
@@ -100,11 +101,23 @@ int util_audit_drop_permissions(uint32_t uid, uint32_t gid) {
  *
  * Return: 0 on success, or a negative error code on failure.
  */
-int util_audit_log(const char *message, uid_t uid) {
-        int r;
+int util_audit_log(int type, const char *message, uid_t uid) {
+        int r, audit_type;
+
+        switch(type) {
+                case AUDIT_TYPE_AVC:
+                        audit_type = AUDIT_USER_AVC;
+                        break;
+                case AUDIT_TYPE_SELINUX_ERROR:
+                        audit_type = AUDIT_USER_SELINUX_ERR;
+                        break;
+                case AUDIT_TYPE_NOAUDIT:
+                default:
+                        audit_type = 0;
+        }
 
-        if (audit_fd >= 0) {
-                r = audit_log_user_avc_message(audit_fd, AUDIT_USER_AVC, message, NULL, NULL, NULL, uid);
+        if (audit_fd >= 0 && audit_type > 0) {
```

Comment:
Maybe be more explicit and use `type != AUDIT_TYPE_NOAUDIT`? I am a bit unsure about relying on external enums to not be 0 (especially since I did not find an explicit statement in libaudit that guarantees such ABI).

Reply:
That would work if something comes in as `AUDIT_TYPE_NOAUDIT`, but if a type isn't an enum value, then this wouldn't work. Maybe set `audit_type` to `-1` for AUDIT_TYPE_NOAUDIT/default?

Reply:
Why would anyone call this with an invalid enum value? This is our internal API, and we control all callers. They only use values we define, right?

Reply:
My mode is always to have defensive programming. If you feel this is overkill I get it in this case.